src/int_32.ts

```diff
@@ -48,11 +49,7 @@ export class Int32 extends BSONValue {
    * @param value - the string we want to represent as an int32.
    */
   static fromString(value: string): Int32 {
-    const cleanedValue = !/[^0]+/.test(value)
-      ? value.replace(/^0+/, '0') // all zeros case
-      : value[0] === '-'
-        ? value.replace(/^-0+/, '-') // negative number with leading zeros
-        : value.replace(/^\+?0+/, ''); // positive number with leading zeros
+    const cleanedValue = removeLeadingZeros(value);
```

Comment:
```
BSON.Int32.fromString('+32')
Uncaught BSONError: Input: '+32' is not a valid Int32 string
```
Same issue in Long, is this intentional? I think it should be permitted

Reply:
This is not intentional, it is a mistake on my end. Good thing that editing the shared function will fix the problem for both!